core/store/src/db/rocksdb.rs

```diff
@@ -415,24 +407,74 @@ impl Database for RocksDB {
         }
     }
 
-    fn create_checkpoint(&self, path: &std::path::Path) -> anyhow::Result<()> {
+    fn create_checkpoint(
+        &self,
+        path: &std::path::Path,
+        columns_to_keep: Option<&[DBCol]>,
+    ) -> anyhow::Result<()> {
         let _span =
             tracing::info_span!(target: "state_snapshot", "create_checkpoint", ?path).entered();
         let cp = ::rocksdb::checkpoint::Checkpoint::new(&self.db)?;
-        cp.create_checkpoint(path)?;
+        cp.create_checkpoint(path)
+            .with_context(|| format!("failed to create checkpoint at {}", path.display()))?;
+
+        if let Some(columns_to_keep) = columns_to_keep {
+            let opts = common_rocksdb_options();
+            let cfs = cf_descriptors(
+                &DBCol::iter().collect::<Vec<_>>(),
+                &StoreConfig::default(),
+                Temperature::Hot,
+            );
+            let mut db = DB::open_cf_descriptors(&opts, path, cfs)
+                .with_context(|| format!("failed to open checkpoint at {}", path.display()))?;
+            for col in DBCol::iter() {
+                if !columns_to_keep.contains(&col) {
+                    if col == DBCol::DbVersion {
```

Comment:
mini nit: To make the for loop less packed perhaps you can move that check before the loop and always add db version to the `columns_to_keep`?

Reply:
hmm yea normally i would like doing this kind of thing to make loops less nested and packed but here the columns_to_keep arg is just a `&[]` (used to be a Vec before this PR but that didnt make a lot of sense to me), so cant easily add an element to it